eloquentjs/4_4_deep_comparison.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * Performs deep comparison of two types and determines equality
+ * @param {*} itemA 
+ * @param {*} itemB 
+ * @returns 
+ */
+function deepEqual(itemA, itemB) {
+    if(itemA === null && itemB === null) {
+        return true;
+    }
+    if (typeof itemA === typeof itemB) {
+        if (typeof itemA === 'object') {
+            if(itemA === null || itemB === null) {
+                return false;
+            }
+
+            let keysA = Object.entries(itemA);
+            let keysB = Object.entries(itemB);
+
+            if (keysA.length !== keysB.length) {
+                return false;
+            }
+
+            if (keysA.length === 0) {
+                throw "objects being compared have no enumerable data types";
+            }
+            
+            for (let i = 0; i < keysA.length; i++) {
```

Comment:
The ordering behavior described in [this](https://dev.to/frehner/the-order-of-js-object-keys-458d) article applies to Object.entries as well. Can you figure out what's wrong with your implementation?

Reply:
Oh beans. That's pretty devious. If I found the same problem I believe its this : 
```
let testCarA = {3: ['a','b','c']}; // numerical 3
let testCarB = {'3': ['a','b','c']}; // string 3
console.log(deepEqual(testCarA, testCarB)); //returns true 🤦‍♂️
```